packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -44,14 +96,38 @@ impl<C: AsClient> crud::Read<PersistedEntity> for PostgresStore<C> {
                 let created_by_id = row.get(5);
                 let updated_by_id = row.get(6);
 
+                let link_metadata = {
+                    let left_entity_id: Option<EntityId> = row.get(7);
+                    let right_entity_id: Option<EntityId> = row.get(8);
+                    match (left_entity_id, right_entity_id) {
+                        (Some(left_entity_id), Some(right_entity_id)) => {
+                            Some(LinkEntityMetadata::new(
+                                left_entity_id,
+                                right_entity_id,
+                                row.get(9),
+                                row.get(10),
+                            ))
+                        }
+                        (None, None) => None,
+                        (Some(_), None) => unreachable!(
+                            "It's not possible to have a link entity with only the left entity id \
+                             specified"
+                        ),
+                        (None, Some(_)) => unreachable!(
+                            "It's not possible to have a link entity with only the right entity \
+                             id specified"
+                        ),
```

Comment:
Are these not combined so we can have different text in the unreachables?

Reply:
I also think it's nice to actually have it explicit in this case.